### PR TITLE
chore(Coder plugin): import preview version of Coder SDK into plugin

### PR DESCRIPTION
**Pull request by Parkreiner**
Scaffolding needed for #114

## Changes made
- Brought in vendored versions of the Coder core files

## Notes
- For the most part, it can be assumed that all vendored files come directly [from Coder core's `api` section](https://github.com/coder/coder/tree/main/site/src/api)
   - Had to make a few changes to satisfy Backstage's ESLint rules, but I'm going to try highlighting the important parts. Otherwise, you can ignore `api.ts` and `typesGenerated.ts`
   - In practice, this PR is like 70 lines long
   - Though I do think there are a few things that the linter caught that are worth updating in core. Will try to make a PR soon

### CODE REVIEW

File: plugins/backstage-plugin-coder/src/api/vendoredSdk/index.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+// whether they pollute Intellisense when someone is using the SDK. Most of
+// these properties don't deal with APIs and are mainly helpers in Core
+type PropertyToHide =
+  | 'getJFrogXRayScan'
+  | 'getCsrfToken'
+  | 'setSessionToken'
+  | 'setHost'
+  | 'getAvailableExperiments'
+  | 'login'
+  | 'logout';
```
[the remark this answers is not in the document]
Comment (code-asher) [reply plugins/backstage-plugin-coder/src/api/vendoredSdk/index.ts:21]: Looks good to me although to be honest I did not spend a whole lot of time looking through everything we are exporting.

I wonder though if we want to move these to an `InternalApiMethods` or something so we will not have to filter them out here.

[the remark this answers is not in the document]
Comment (code-asher) [reply plugins/backstage-plugin-coder/src/api/vendoredSdk/index.ts:21]: I went through it a bit more after all haha

`convertToOAUTH` probably would make sense to hide as well.

I am not sure about `waitForBuild`.  It is a helper function, not a straight API call.  Maybe we exclude helpers for now but can think about exposing them differently.

Similarly we have a bunch of convenience wrappers around the `postWorkspaceBuild` endpoint.  At the very least I think we should expose them differently, if at all (the Go sdk does not have equivalents for example, as far as I know).

`addMember` and `removeMember` are also wrappers/helpers that do not exist on the Go sdk.

`changeWorkspaceVersion`, `updateWorkspace`, these are somewhat bespoke for our dashboard, not necessarily that they would not be useful for users, but not really API methods either.

`getWorkspaceParameters` seems to just combine two other calls.

**Review by code-asher — APPROVED** [review]
Exciting!

File: plugins/backstage-plugin-coder/src/api/vendoredSdk/index.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (Parkreiner) [reply plugins/backstage-plugin-coder/src/api/vendoredSdk/index.ts:21]: Yeah, I would normally agree that it'd be better to hide these things, but my hope with the update process was that you could literally copy-and-paste the files from core with zero changes, and have things just work

I'm going to add the methods you highlighted, but keep everything else the same for now. If we're going to need better separation, I feel like those changes should be made in core

[the remark this answers is not in the document]
Comment (code-asher) [reply plugins/backstage-plugin-coder/src/api/vendoredSdk/index.ts]: Oh yeah 100% agree those changes should be made in core, not here.

**Pull request merged**